aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -118,6 +118,65 @@ TORCH_META_FUNC(argmin)
 
 } // namespace meta
 
+namespace meta {
+
+TORCH_META_FUNC(minmax)
+(const Tensor& self, c10::optional<int64_t> dim_opt, bool keepdim) {
+  DimVector shape;
+  if (dim_opt.has_value()) {
+    auto dim = maybe_wrap_dim(dim_opt.value(), self.ndimension());
+    native::zero_numel_check_dims(self, dim, "minmax");
+    shape = get_reduction_shape(self, dim, keepdim);
+  } else if (keepdim) {
```

Comment:
what happens if `dim_opt` is empty and `keepdim` is false?

Reply:
The same wrong behavior that most reduction ops have now, it will reduce everything. Good catch! I'll update it to a noop.

Reply:
But it shouldn't be a no-op. `x.minmax()` (which is empty dim_opt and `false` keepdim) should reduce all dimensions. It's just with your logic it never checks for `self.numel() > 0` and doesn't set 0-d tensor `shape` (I don't know if `set_output` can deal with uninitialized DimeVector argument).

Reply:
Oh sorry, I misunderstood you. Yes, you are right! empty dim_opt is when `dim=None` which should reduce all dimensions. `dim=[]` should be a noop. Thanks for catching this, I'll update it accordingly.